src/snackbar/snackbar-container.svelte

```diff
@@ -0,0 +1,46 @@
+<script>
+  import { setContext } from 'svelte';
+  import Snackbar from './snackbar.svelte';
+  import snackbarContextKey from './snackbar-context-key.js';
+
+  export let position;
+  let registeredSnackbars = new Set();
+
+  function removeSnackbar(key) {
+    registeredSnackbars.delete(key);
+    registeredSnackbars = registeredSnackbars;
+  }
+
+  function showSnackbar(options) {
+    const {
+      component = Snackbar,
+      props = {},
+      duration = 3000,
+    } = options;
+
+    const key = { component, props };
+    key.timeoutID = setTimeout(removeSnackbar, duration, key);
+    registeredSnackbars.add(key);
+    registeredSnackbars = registeredSnackbars;
+
+    return function close() {
+      clearTimeout(key.timeoutID);
+      registeredSnackbars.delete(key);
+      registeredSnackbars = registeredSnackbars;
```

Comment:
Isn't this exactly `removeSnackbar(key)`? DRY

Reply:
It's also clearing the timeout and closure-ing the key. There has to be a new function object anyway, I see no reason to replace two lines for one in some weird way

Reply:
I meant just these 2 lines, not the `close` function. The reason to use `removeSnackbar` inside it would be to avoid repetition and changing the same logic in only one place

Reply:
I know, but these two lines really don't seem worth wrapping in a function call to me

Reply:
It's not wrapping, they are already in a function. It might not be worth it now, but future refactoring will be easier if we follow the DRY principle

Reply:
I still don't think it's worth it, but I've run out of energy for this issue, so whatever